enclave-runtime/src/tls_ra/tls_ra_client.rs

```diff
@@ -73,17 +74,20 @@ where
 	///
 	/// We trust here that the server sends us the correct data, as
 	/// we do not have any way to test it.
-	fn read_shard(&mut self) -> EnclaveResult<()> {
-		debug!("read_shard called, about to call self.write_shard().");
-		self.write_shard()?;
-		debug!("self.write_shard() succeeded.");
+	fn obtain_provisioning_for_shard(&mut self, account: AccountId) -> EnclaveResult<()> {
```

Comment:
Maybe better to call it: `request_provisioning_for_shard`?

Reply:
thought about it, but "request" is the action in the subsequent call